src/main/java/it/pagopa/pn/deliverypush/service/impl/ConfidentialInformationServiceImpl.java

```diff
@@ -89,45 +81,37 @@ private ConfidentialTimelineElementDtoInt getConfidentialDtoFromTimeline(Timelin
 
     @Override
     public Optional<ConfidentialTimelineElementDtoInt> getTimelineElementConfidentialInformation(String iun, String timelineElementId) {
-        ResponseEntity<ConfidentialTimelineElementDto> resp = pnDataVaultClient.getNotificationTimelineByIunAndTimelineElementId(iun, timelineElementId);
+      ConfidentialTimelineElementDto dtoExt = pnDataVaultClient.getNotificationTimelineByIunAndTimelineElementId(iun, timelineElementId);
 
-        if (resp.getStatusCode().is2xxSuccessful()) {
-            log.debug("getTimelineElementConfidentialInformation OK for - iun {} timelineElementId {}", iun, timelineElementId);
-            ConfidentialTimelineElementDto dtoExt = resp.getBody();
 
-            if (dtoExt != null) {
-                return Optional.of(ConfidentialTimelineElementDtoMapper.externalToInternal(dtoExt));
-            }
+      log.debug("getTimelineElementConfidentialInformation OK for - iun {} timelineElementId {}", iun, timelineElementId);
+         
 
-            log.debug("getTimelineElementConfidentialInformation haven't confidential information for - iun {} timelineElementId {}", iun, timelineElementId);
-            return Optional.empty();
-        } else {
-            log.error("getTimelineElementConfidentialInformation Failed for - iun {} timelineElementId {}", iun, timelineElementId);
-            throw new PnInternalException("getTimelineElementConfidentialInformation Failed for - iun " + iun + " timelineElementId " + timelineElementId, ERROR_CODE_DELIVERYPUSH_TIMELINECONFIDENTIALFAILED);
```

Comment:
In tutti i casi, non solo questo, non mi torna che siano state eliminate le gestione delle casistiche di errore. Mi sarei aspettato in questi casi non più il check sul 200 come ci siamo detti, ma un try catch sull'exception PnHttpResponseException (rilanciata dal RestTemplateResponseErrorHandler), seguendo la logica attualmente presente, rilanciare la PnInternalException con CODE specifico ecc. Nei casi in cui sono previste gestioni specifiche per alcuni status code intercettare, ottenerli dall'exception ed agire di conseguenza.
NOTA: E' inoltre importatissimo riuscire ad effettuare dei Test in locale prima di rilasciare tale modifica, per essere sicuri che la logica di cui abbiamo discusso, funzioni effettivamente come l'abbiamo immaginata.

Reply:
Ovviamente la logica di avere le exception rilanciate, con codice specifico definito, ci permette di avere una situazione più chiara e semplice da individuare a livello di log e soprattuto con le exception specifiche e i codici, riusciamo a restituire informazioni corrette con status code specifici alle API esposte all'esterno

Reply:
I blocchi di codice nell'else non venivano mai eseguiti, proprio perchè c'è  il RestTemplateResponseErrorHandler che se la response è diversa da 2XX lancia l'eccezione PnHttpResponseException. 
PnHttpResponseException all'interno ha lo stato http.  Nel codice non c'è nessuna logica sullo stato http. 

IL merge non lo faccio. Ti ho chiesto la PL visto che  siamo partiti da una richiesta e ne stiamo lavorando un 'altra accordata verbalmente.

Reply:
Sisi mi torna che i blocchi else non venivamo mai eseguiti, l'obbiettivo della issue era quello. Ad un certo punto ci siamo accortit che li non arrivava mai e dunque bisognava trovare una soluzione che mitigasse il problema e dunque lanciasse l'exception specifiche nei casi di NOT 200, piuttosto che altri tipi di exception in 400 se previsti ecc. Questo sarebbe possibile con la soluzione accordata dal momento che PnHttpResponseException come detto ha l'informazione sullo stato HTTP.
Comunque quando sei libero, contattami, che definiamo meglio la cosa

Reply:
Risolta: viene propagata la PnHttpResponseException che extender l'internal, dunque verrà catturata e gestita correttamente dall'handler come exception generica.